### ISSUE TITLE
//move and //undo Problem

### ISSUE THREAD

**Issue by AntZaro** [SBPrime/AsyncWorldEdit-Premium#189]
I'm running version 3.6.8 with worldedit 7.0.1 on a 1.14.4 server. Each time this player attempts to use //move, he cannot undo it. Async gives an error. In other words, it moves fine. But nothing happens when he does //undo, it swallows whatever change he attempted. 

Log: 

[17:52:45] [Server thread/INFO]: TheCrimsonChin issued server command: //move 1 down
[17:52:59] [Server thread/INFO]: TheCrimsonChin issued server command: //undo
[17:52:59] [Craft Scheduler Thread - 393/WARN]: [AsyncWorldEdit] Plugin AsyncWorldEdit v3.6.8 generated an exception while executing task 5003
java.lang.NullPointerException: null
	at com.sk89q.worldedit.util.collection.LocatedBlockList.get(LocatedBlockList.java:70) ~[?:?]
	at com.sk89q.worldedit.extent.reorder.MultiStageReorder.lambda$getBufferedBlock$12(MultiStageReorder.java:258) ~[?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:1.8.0_144]
	at java.util.HashMap$ValueSpliterator.tryAdvance(Unknown Source) ~[?:1.8.0_144]
	at java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source) ~[?:1.8.0_144]
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source) ~[?:1.8.0_144]
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:1.8.0_144]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:1.8.0_144]
	at java.util.stream.FindOps$FindOp.evaluateSequential(Unknown Source) ~[?:1.8.0_144]
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:1.8.0_144]
	at java.util.stream.ReferencePipeline.findAny(Unknown Source) ~[?:1.8.0_144]
	at com.sk89q.worldedit.extent.reorder.MultiStageReorder.getBufferedBlock(MultiStageReorder.java:260) ~[?:?]
	at com.sk89q.worldedit.extent.AbstractBufferingExtent.getBlock(AbstractBufferingExtent.java:54) ~[?:?]
	at com.sk89q.worldedit.extent.reorder.MultiStageReorder.setBlock(MultiStageReorder.java:223) ~[?:?]
	at com.sk89q.worldedit.history.change.BlockChange.undo(BlockChange.java:89) ~[?:?]
	at org.primesoft.asyncworldedit.worldedit.UndoProcessor.resume(UndoProcessor.java:119) ~[?:?]
	at com.sk89q.worldedit.function.operation.Operations.__aee2e42fdbb1466bbd01d912f432015a__completeBlindly(Operations.java:74) ~[?:?]
	at com.sk89q.worldedit.function.operation.Operations_InnerForMethod__aee2e42fdbb1466bbd01d912f432015a__completeBlindly.execute(Unknown Source) ~[?:?]
	at org.primesoft.asyncworldedit.asyncinjector.async.AsyncOperationProcessor.process(AsyncOperationProcessor.java:181) ~[?:?]
	at org.primesoft.asyncworldedit.injector.core.visitors.Helpers.executeMethod(Helpers.java:85) ~[?:?]
	at com.sk89q.worldedit.function.operation.Operations.completeBlindly(Operations.java) ~[?:?]
	at org.primesoft.asyncworldedit.worldedit.UndoProcessor.processUndo(UndoProcessor.java:83) ~[?:?]
	at org.primesoft.asyncworldedit.worldedit.CancelabeEditSession.doUndo(CancelabeEditSession.java:415) ~[?:?]
	at org.primesoft.asyncworldedit.worldedit.CancelabeEditSession.undo(CancelabeEditSession.java:411) ~[?:?]
	at org.primesoft.asyncworldedit.worldedit.AsyncEditSession$1.task(AsyncEditSession.java:154) ~[?:?]
	at org.primesoft.asyncworldedit.worldedit.AsyncTask.doRun(AsyncTask.java:71) ~[?:?]
	at org.primesoft.asyncworldedit.worldedit.BaseTask.run(BaseTask.java:160) ~[?:?]
	at org.primesoft.asyncworldedit.utils.SchedulerUtils.processTaskQueue(SchedulerUtils.java:168) ~[?:?]
	at org.primesoft.asyncworldedit.utils.SchedulerUtils.access$000(SchedulerUtils.java:62) ~[?:?]
	at org.primesoft.asyncworldedit.utils.SchedulerUtils$1.run(SchedulerUtils.java:131) ~[?:?]
	at org.bukkit.craftbukkit.v1_14_R1.scheduler.CraftTask.run(CraftTask.java:84) ~[patched_1.14.4.jar:git-Paper-148]
	at org.bukkit.craftbukkit.v1_14_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:54) ~[patched_1.14.4.jar:git-Paper-148]
	at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[patched_1.14.4.jar:git-Paper-148]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:1.8.0_144]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:1.8.0_144]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_144]
[17:53:03] [Server thread/INFO]: TheCrimsonChin issued server command: //move 2 down
[17:53:07] [Server thread/INFO]: TheCrimsonChin issued server command: //undo
[17:53:08] [Server thread/INFO]: TheCrimsonChin issued server command: //undo

**Comment by SBPrime** [comment]
Can you send me the screen of blocks that you are trying to move?

**Comment by SBPrime** [comment]
Also could you send me the exact version of WE and the server startup log?
In addition (optional) when you do a next restart please create "classes" directory (in the root of the server) and send me com.sk89q.worldedit.extent.reorder.MultiStageReorder.class file.

**Comment by AntZaro** [comment]
![Screen Shot 2019-07-29 at 21 33 51](https://user-images.githubusercontent.com/17606915/62093801-9b09a200-b248-11e9-9f87-54e3b7efcb54.png)
![Screen Shot 2019-07-29 at 21 34 02](https://user-images.githubusercontent.com/17606915/62093806-9f35bf80-b248-11e9-8867-406943f70e77.png)

**Comment by AntZaro** [comment]
WorldEdit version 7.0.1-rc1;edb2e91
```
[16:33:59] [Server thread/INFO]: Starting minecraft server version 1.14.4
[16:33:59] [Server thread/INFO]: Loading properties
[16:33:59] [Server thread/INFO]: This server is running Paper version git-Paper-148 (MC: 1.14.4) (Implementing API version 1.14.4-R0.1-SNAPSHOT)
[16:33:59] [Server thread/INFO]: Server Ping Player Sample Count: 12
[16:33:59] [Server thread/INFO]: Using 4 threads for Netty based IO
[16:33:59] [Server thread/INFO]: Debug logging is disabled
[16:33:59] [Server thread/INFO]: Default game type: SURVIVAL
[16:33:59] [Server thread/INFO]: Generating keypair
[16:34:00] [Server thread/INFO]: Starting Minecraft server on 127.0.0.1:25568
[16:34:00] [Server thread/INFO]: Using epoll channel type
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Injector platform set to: InjectorBukkit
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Injecting NMS classes...
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Injecting WorldEdit classes...
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.util.eventbus.EventBus
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.math.BlockVector2
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.math.BlockVector3
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.math.Vector2
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.math.Vector3
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.world.block.BlockStateHolder
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.world.block.BaseBlock
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.world.block.BlockState
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.entity.BaseEntity
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.util.Location
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.world.biome.BiomeType
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.world.weather.WeatherType
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.EditSession
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.function.operation.Operations
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.extent.clipboard.BlockArrayClipboard
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.extension.platform.PlayerProxy
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.extent.reorder.MultiStageReorder
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.regions.AbstractRegion
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.regions.EllipsoidRegion
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.regions.ConvexPolyhedralRegion
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.regions.TransformRegion
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.regions.RegionIntersection
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.regions.NullRegion
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.regions.Polygonal2DRegion
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.regions.CylinderRegion
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.regions.CuboidRegion
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.command.SnapshotUtilCommands
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.command.ScriptingCommands
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.command.SchematicCommands
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.command.RegionCommands
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.command.UtilityCommandsRegistration
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Modiffy class com.sk89q.worldedit.command.RegionCommandsRegistration
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Creating class org/primesoft/asyncworldedit/worldedit/entity/PlayerWrapper
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Creating class org/primesoft/asyncworldedit/worldedit/entity/NoPermsPlayer
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Creating class org/primesoft/asyncworldedit/worldedit/extension/platform/NoPermsActor
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Creating class org/primesoft/asyncworldedit/injector/injected/entity/IPlayerFactory
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Creating class org/primesoft/asyncworldedit/injector/injected/extension/platform/IActorFactory
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] Injector Enabled
[16:34:03] [Server thread/INFO]: [AsyncWorldEdit] New class factory set to: org.primesoft.asyncworldedit.asyncinjector.async.AsyncClassFactory
[16:34:03] [Server thread/INFO]: [WorldEdit] Loading WorldEdit v7.0.1-rc1;edb2e91
[16:34:04] [Server thread/INFO]: Got request to register class com.sk89q.worldedit.bukkit.BukkitServerInterface with WorldEdit [com.sk89q.worldedit.extension.platform.PlatformManager@215db34a]
[16:34:05] [Server thread/INFO]: [WorldEdit] Enabling WorldEdit v7.0.1-rc1;edb2e91
[16:34:05] [Server thread/INFO]: WEPIF: Using the Bukkit Permissions API.
[16:34:06] [Server thread/INFO]: Using com.sk89q.worldedit.bukkit.adapter.impl.Spigot_v1_14_R4 as the Bukkit adapter
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] Enabling AsyncWorldEdit v3.6.8
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] Warning: Unable to get class com.sk89q.worldedit.extent.reorder.MultiStageReorder$Stage3Committer
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] Warning: Unable to get class com.sk89q.worldedit.extent.reorder.MultiStageReorder$Stage3Committer
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] Platform set to Bukkit API
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] ===================================================================================
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] =     ___                         _       __           __    ________    ___ __   =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] =    /   |  _______  ______  ____| |     / /___  _____/ /___/ / ____/___/ (_) /_  =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] =   / /| | / ___/ / / / __ \/ ___/ | /| / / __ \/ ___/ / __  / __/ / __  / / __/  =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] =  / ___ |(__  ) /_/ / / / / /__ | |/ |/ / /_/ / /  / / /_/ / /___/ /_/ / / /_    =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] = /_/  |_/____/\__, /_/ /_/\___/ |__/|__/\____/_/  /_/\__,_/_____/\__,_/_/\__/    =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] =             /____/                                                              =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] = Author: SBPrime                                                                 =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] =                                                                                 =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] ===================================================================================
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] Initializing WorldEdit.
[16:34:06] [AWE:Cron:undoCleanup/INFO]: [AsyncWorldEdit] Undo cleanup started...
[16:34:06] [AWE:Cron:undoCleanup/INFO]: [AsyncWorldEdit] ...undo cleanup done.
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] No matching adapter found for Bukkit.v1_14_R1, DirectChunkAPI disabled
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] PlotMeFix set to class org.primesoft.asyncworldedit.plotme.NullFix.
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] Enabled
```

**Comment by AntZaro** [comment]
`[16:34:05] [Server thread/INFO]: [WorldEdit] Enabling WorldEdit v7.0.1-rc1;edb2e91
[16:34:05] [Server thread/INFO]: WEPIF: Using the Bukkit Permissions API.
[16:34:06] [Server thread/INFO]: Using com.sk89q.worldedit.bukkit.adapter.impl.Spigot_v1_14_R4 as the Bukkit adapter
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] Enabling AsyncWorldEdit v3.6.8
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] Warning: Unable to get class com.sk89q.worldedit.extent.reorder.MultiStageReorder$Stage3Committer
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] Warning: Unable to get class com.sk89q.worldedit.extent.reorder.MultiStageReorder$Stage3Committer
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] Platform set to Bukkit API
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] ===================================================================================
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] =     ___                         _       __           __    ________    ___ __   =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] =    /   |  _______  ______  ____| |     / /___  _____/ /___/ / ____/___/ (_) /_  =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] =   / /| | / ___/ / / / __ \/ ___/ | /| / / __ \/ ___/ / __  / __/ / __  / / __/  =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] =  / ___ |(__  ) /_/ / / / / /__ | |/ |/ / /_/ / /  / / /_/ / /___/ /_/ / / /_    =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] = /_/  |_/____/\__, /_/ /_/\___/ |__/|__/\____/_/  /_/\__,_/_____/\__,_/_/\__/    =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] =             /____/                                                              =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] = Author: SBPrime                                                                 =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] =                                                                                 =
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] ===================================================================================
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] Initializing WorldEdit.
[16:34:06] [AWE:Cron:undoCleanup/INFO]: [AsyncWorldEdit] Undo cleanup started...
[16:34:06] [AWE:Cron:undoCleanup/INFO]: [AsyncWorldEdit] ...undo cleanup done.
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] No matching adapter found for Bukkit.v1_14_R1, DirectChunkAPI disabled
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] PlotMeFix set to class org.primesoft.asyncworldedit.plotme.NullFix.
[16:34:06] [Server thread/INFO]: [AsyncWorldEdit] Enabled
`

**Comment by AntZaro** [comment]
I hope that covers it... sorry about the messy formatting on the log paste.

**Comment by SBPrime** [comment]
hi i managed to reproduce it I should push a fix within 24h
I also fixed the log :)

**Comment by SBPrime** [comment]
Fixed in 3.6.9
